source/common/http/conn_manager_impl.cc

```diff
@@ -812,12 +814,16 @@ ConnectionManagerImpl::ActiveStream::ActiveStream(ConnectionManagerImpl& connect
   if (connection_manager_.config_.accessLogFlushInterval().has_value()) {
     access_log_flush_timer_ =
         connection_manager.read_callbacks_->connection().dispatcher().createTimer([this]() -> void {
-          // If the request is complete, we've already done the stream-end access-log, and shouldn't
-          // do the periodic log.
-          if (!streamInfo().requestComplete().has_value()) {
-            filter_manager_.log();
+          if (!streamInfo().streamState() ||
+              streamInfo().streamState() != StreamInfo::StreamState::Ended) {
             refreshAccessLogFlushTimer();
           }
+
+          // If the request is not in active state, we shouldn't do the periodic log.
+          if (streamInfo().streamState() &&
+              streamInfo().streamState() == StreamInfo::StreamState::InProgress) {
```

Comment:
@alyssawilk, I now recall that one of the usages of the stream state were to determine whether it's required to log() on periodic access logs here. Considering that you requested to delete it, is there another valid alternative to determine whether it should be logged here?

Reply:
I'll revert to how it was before this change - only stop periodic logs if requestComplete is set